Tests/src/PublicAPITests/HotSpotSequenceSystemTests.cpp

```diff
@@ -0,0 +1,648 @@
+{
+	auto [Result] = Awaitable(&csp::systems::HotSpotSequenceSystem::DeleteHotspotGroup, HotSpotSequenceSystem, GroupName).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), ExpectedResultCode);
+	EXPECT_EQ(Result.GetFailureReason(), ExpectedResultFailureCode);
+}
+
+void GetHotpotGroup(csp::systems::HotSpotSequenceSystem* HotSpotSequenceSystem,
+					const csp::common::String& GroupName,
+					csp::systems::HotSpotGroup& Group,
+					csp::systems::EResultCode ExpectedResultCode				  = csp::systems::EResultCode::Success,
+					csp::systems::ERequestFailureReason ExpectedResultFailureCode = csp::systems::ERequestFailureReason::None)
+{
+	auto [Result] = Awaitable(&csp::systems::HotSpotSequenceSystem::GetHotspotGroup, HotSpotSequenceSystem, GroupName).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), ExpectedResultCode);
+	EXPECT_EQ(Result.GetFailureReason(), ExpectedResultFailureCode);
+	csp::systems::HotSpotGroup group = Result.GetHotSpotGroup();
+	if (Result.GetResultCode() == csp::systems::EResultCode::Success)
+	{
+		Group = group;
+	}
+}
+
+void UpdateHotspotGroup(csp::systems::HotSpotSequenceSystem* HotSpotSequenceSystem,
+						const csp::common::String& GroupName,
+						const csp::common::Array<csp::common::String>& Items,
+						csp::systems::HotSpotGroup& HotSpotGroup,
+						csp::systems::EResultCode ExpectedResultCode				  = csp::systems::EResultCode::Success,
+						csp::systems::ERequestFailureReason ExpectedResultFailureCode = csp::systems::ERequestFailureReason::None)
+{
+	auto [Result]
+		= Awaitable(&csp::systems::HotSpotSequenceSystem::UpdateHotspotGroup, HotSpotSequenceSystem, GroupName, Items).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), ExpectedResultCode);
+	EXPECT_EQ(Result.GetFailureReason(), ExpectedResultFailureCode);
+
+	if (ExpectedResultCode == csp::systems::EResultCode::Success)
+	{
+		csp::systems::HotSpotGroup group = Result.GetHotSpotGroup();
+
+		EXPECT_EQ(group.Items.Size(), Items.Size());
+
+		for (int i = 0; i < group.Items.Size(); ++i)
+		{
+			EXPECT_EQ(group.Items[i], Items[i]);
+		}
+
+		HotSpotGroup = group;
+	}
+}
+
+void RenameHotspotGroup(csp::systems::HotSpotSequenceSystem* HotSpotSequenceSystem,
+						const csp::common::String& GroupName,
+						const csp::common::String& NewGroupName,
+						csp::systems::HotSpotGroup& HotSpotGroup,
+						csp::systems::EResultCode ExpectedResultCode				  = csp::systems::EResultCode::Success,
+						csp::systems::ERequestFailureReason ExpectedResultFailureCode = csp::systems::ERequestFailureReason::None)
+{
+	auto [Result]
+		= Awaitable(&csp::systems::HotSpotSequenceSystem::RenameHotspotGroup, HotSpotSequenceSystem, GroupName, NewGroupName).Await(RequestPredicate);
+
+	EXPECT_EQ(Result.GetResultCode(), ExpectedResultCode);
+	EXPECT_EQ(Result.GetFailureReason(), ExpectedResultFailureCode);
+
+	if (ExpectedResultCode == csp::systems::EResultCode::Success)
```

Comment:
Verify the new name matches the requested name

Reply:
We check metadata tags now for this